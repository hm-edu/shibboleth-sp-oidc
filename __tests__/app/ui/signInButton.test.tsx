import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';
import { cleanup, fireEvent, render, screen } from '@testing-library/react';
import SignInButton from '@/app/ui/signInButton';

describe('SignInButton', () => {
  beforeEach(() => {
    render(SignInButton());
  });

  it('should exists', () => {
    expect(screen.getByRole('button')).toBeDefined();
  });

  it('should match snapshot', () => {
    expect(screen.getByRole('button')).toMatchSnapshot();
  });

  it('should trigger signOut function', () => {
    const { mockedSignIn } = vi.hoisted(() => {
      return { mockedSignIn: vi.fn(() => Promise.resolve()) };
    });
    vi.mock('next-auth/react', () => {
      return { signIn: mockedSignIn };
    });

    fireEvent.click(screen.getByText(/Anmelden/));
    expect(mockedSignIn).toHaveBeenCalledOnce();
  });

  afterEach(() => {
    cleanup();
  });
});
