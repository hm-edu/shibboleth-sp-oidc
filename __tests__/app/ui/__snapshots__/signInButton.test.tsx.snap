// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SignInButton > should match snapshot 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorInherit css-sbfmij-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  Anmelden
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
